mriqc/viz/fmriplots.py

```diff
@@ -135,24 +135,34 @@ def fmricarpetplot(func_data, segmentation, outer_gs, tr=None, nskip=0):
 
     # Carpet plot
     ax1 = plt.subplot(gs[1])
-    ax1.imshow(detrended[order, :], interpolation='nearest',
+
+    # Avoid segmentation faults for long acquisitions by decimating the input data
+    long_cutoff = 800
+    if detrended.shape[1] > long_cutoff:
+        data = detrended[order, ::2]
+    else:
+        data = detrended[order, :]
+
+    ax1.imshow(data, interpolation='nearest',
```

Comment:
Actually, the size of the output file does not change much - the seg fault must be caused by internal representation. I did some memory profiling and lack of RAM wasn't to blame.

I don't think `extent` would work since it would just crop the plot (and we want to scale it). I tried changing the `dpi`, but that only worked when using `.png` not `.svg` - not quite sure why.